### PR TITLE
Catch the JsonError about multi-tile more gracefully

### PR DESCRIPTION
**Pull request by ToxiClay**
#### Summary
None

#### Purpose of change

~~This PR includes a brief comment on how to fix the multitile JSON error, as it is not an uncommon occurrence for people to ask. Including resolution instructions in the error itself is much more user-friendly.~~

This PR more gracefully handles the error message concerning an improperly-set flag, allowing the game to load and emitting a debug message instead of blocking.

#### Describe the solution

I worded the words.

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
This error does not occur solely because people are using out of date tilesets and need to update it. Sure, it's the most player facing error, but error messages should not be prescribing certain behaviour unless that behaviour is always the solution to the problem.

If you want to make this less troublesome to users, you can throw then immediately catch and emit a debugmsg for this error, instead of throwing an error which makes it all the way up and prevents loading.

**Comment by ToxiClay** [comment]
> This error does not occur solely because people are using out of date tilesets and need to update it. Sure, it's the most player facing error, but error messages should not be prescribing certain behaviour unless that behaviour is always the solution to the problem.
> 
> If you want to make this less troublesome to users, you can throw then immediately catch and emit a debugmsg for this error, instead of throwing an error which makes it all the way up and prevents loading.

Oh. Huh. Sorry; I've never seen that not be the issue.

Um...my C++ is rusty at best, but let me poke around a bit and see if I can figure out how to do that...

**Comment by anothersimulacrum** [comment]
Here's an example
https://github.com/CleverRaven/Cataclysm-DDA/blob/cbcfab179d4e7e7f715e21d44816d1bee1132868/src/assign.h#L44-L50

**Comment by ToxiClay** [comment]
Huh. I can catch a `JsonError` even though the return type of `throw_error()` is `void`?

File: src/cata_tiles.cpp [hunk 1]
```diff
@@ -934,7 +934,11 @@ void tileset_loader::load_tilejson_from_file( const JsonObject &config )
                     curr_tile.available_subtiles.push_back( s_id );
                 }
             } else if( entry.has_array( "additional_tiles" ) ) {
-                entry.throw_error( "Additional tiles defined, but 'multitile' is not true." );
+                try {
+                        entry.throw_error( "Additional tiles defined, but 'multitile' is not true." );
+                } catch ( const JsonError &err ) {
+                        debugmsg( "(json-error)\n%s", err.what() );
+                }
```
Comment (kevingranade) [inline src/cata_tiles.cpp:941]:
```suggestion
                try {
                    entry.throw_error( "Additional tiles defined, but 'multitile' is not true." );
                } catch( const JsonError &err ) {
                    debugmsg( "(json-error)\n%s", err.what() );
                }
```

**Comment by ToxiClay** [comment]
Oy, it's always the little things that trip me up. Thanks, Kevin.

**Comment by kevingranade** [comment]
I'm super unhppy about throwing and then immediately catching an exception, but at the same time fixing that would be pretty invasive.

**Comment by anothersimulacrum** [comment]
Would something like this work (needs hooking up past JsonIn, to something like a throw_debugmsg?)?
```diff
diff --git a/src/json.cpp b/src/json.cpp
index 8ea59cada1..9709020c56 100644
--- a/src/json.cpp
+++ b/src/json.cpp
@@ -1799,27 +1799,8 @@ std::string JsonIn::line_number( int offset_modifier )
     return ret.str();
 }

-void JsonIn::error( const std::string &message, int offset )
+std::string JsonIn::error_str( const std::string &message, int offset )
 {
-    std::ostringstream err_header;
-    switch( error_log_format ) {
-        case error_log_format_t::human_readable:
-            err_header << "Json error: " << line_number( offset ) << ": ";
-            break;
-        case error_log_format_t::github_action:
-            err_header << "::error " << line_number( offset ) << "::";
-            break;
-    }
-    // if we can't get more info from the stream don't try
-    if( !stream->good() ) {
-        throw JsonError( err_header.str() + escape_data( message ) );
-    }
-    // Seek to eof after throwing to avoid continue reading from the incorrect
-    // location. The calling code of json error methods is supposed to restore
-    // the stream location if it wishes to recover from the error.
-    on_out_of_scope seek_to_eof( [this]() {
-        stream->seekg( 0, std::istream::end );
-    } );
     std::ostringstream err;
     err << message;
     // also print surrounding few lines of context, if not too large
@@ -1891,7 +1872,32 @@ void JsonIn::error( const std::string &message, int offset )
     if( !msg.empty() && msg.back() != '\n' ) {
         msg.push_back( '\n' );
     }
-    throw JsonError( err_header.str() + escape_data( msg ) );
+
+    return escape_data( msg );
+}
+
+void JsonIn::error( const std::string &message, int offset )
+{
+    std::ostringstream err_header;
+    switch( error_log_format ) {
+        case error_log_format_t::human_readable:
+            err_header << "Json error: " << line_number( offset ) << ": ";
+            break;
+        case error_log_format_t::github_action:
+            err_header << "::error " << line_number( offset ) << "::";
+            break;
+    }
+    // if we can't get more info from the stream don't try
+    if( !stream->good() ) {
+        throw JsonError( err_header.str() + escape_data( message ) );
+    }
+    // Seek to eof after throwing to avoid continue reading from the incorrect
+    // location. The calling code of json error methods is supposed to restore
+    // the stream location if it wishes to recover from the error.
+    on_out_of_scope seek_to_eof( [this]() {
+        stream->seekg( 0, std::istream::end );
+    } );
+    throw JsonError( err_header.str() + error_str( message, offset ) );
 }

 void JsonIn::string_error( const std::string &message, const int offset )
diff --git a/src/json.h b/src/json.h
index 550b12086d..4d13fbe43a 100644
--- a/src/json.h
+++ b/src/json.h
@@ -593,6 +593,7 @@ class JsonIn

         // error messages
         std::string line_number( int offset_modifier = 0 ); // for occasional use only
+        std::string error_str( const std::string &message, int offset = 0 );
         [[noreturn]] void error( const std::string &message, int offset = 0 ); // ditto
         // if the next element is a string, throw error after the `offset`th unicode
         // character in the parsed string. if `offset` is 0, throw error right after
```

**Comment by kevingranade** [comment]
Yea something like that, though what I was thinking of was worse.

**Pull request merged**